jobbergate-api/jobbergate_api/apps/job_scripts/services.py

```diff
@@ -49,6 +52,51 @@ class JobScriptFileService(FileService):
         error: Value of type variable "FileModel" of "FileService" cannot be "JobScriptFile"
     """
 
+    async def upsert(
+        self,
+        parent_id: int,
+        filename: str,
+        upload_content: str | bytes | UploadFile,
+        **upsert_kwargs,
+    ) -> FileModel:
+        """
+        Upsert a file instance.
+        """
+        file_type: str = enforce_defined(
+            upsert_kwargs.get("file_type", None),
+            "File type must be defined when upserting a file.",
+            raise_exc_class=ServiceError,
+            raise_kwargs=dict(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY),
+        )
+        await self.validate_entrypoint_file(parent_id, filename, file_type)
+        return await super().upsert(parent_id, filename, upload_content, **upsert_kwargs)
+
+    async def validate_entrypoint_file(self, parent_id: int, filename: str, file_type: FileType | str):
+        """
+        Validate that the entrypoint file is unique.
+        """
+        if file_type != FileType.ENTRYPOINT:
+            return
```

Comment:
It think it would make more sense to put this in the `upsert()` method so that we don't validate at all if it's not an entrypoint file.

Reply:
Do you mean just the if? Or additional logic?

Reply:
Yeah. I mean, if we know outside this function if it's not an entrypoint file, there's no need to call the function.

Reply:
good point, solved in [be1c4a1](https://github.com/omnivector-solutions/jobbergate/pull/305/commits/be1c4a15434f7c75615d9c5316d01a15cd046c5f)